test/integration/TextEditorTest.kt

```diff
@@ -79,15 +80,23 @@ class TextEditorTest : IntegrationTest() {
 
                 Service.client.session.tryOpen(
                     database = testID,
-                    TypeDBSession.Type.DATA
+                    TypeDBSession.Type.SCHEMA
                 )
-                delayAndRecompose(composeRule, Delays.NETWORK_IO)
 
-                Service.schema.reloadEntries()
+                waitUntilTrue(composeRule) {
+                    Service.client.session.type == TypeDBSession.Type.SCHEMA
+                }
+
+                waitUntilTrue(composeRule) {
+                    Service.client.session.typeSchema()!!.contains(commitDateAttributeName)
+                }
 
-                // We can assert that the schema has been written successfully here as the schema
-                // is shown in the type browser.
-                assertNodeExistsWithText(composeRule, text = "commit-date")
+                waitUntilAssertionPasses(composeRule) {
+                    assert(
```

Comment:
Use `assertEquals`. It gives a much more helpful `AssertionError` when it fails.